From 95ad9f452d40f9e2effd05a16b39f4eadd4854b2 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 19 Apr 2022 16:34:51 +0300
Subject: [PATCH] Revert "Revert "[SW-66329] device: add pif arc device""

This patch restores commit 5015baa4a0e3 ("[SW-66329] device: add pif arc
device") that was reverted because the global specs were missing the
latest regs definitions for the pif arc

This reverts commit b4e1cfbfef6c3cf24a675f539df9547d9d292151.

Change-Id: I6d3bcf6423d42239eb473df1b52958e2a7e951bb
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/app/pcie_pif/pcie_pif.c b/app/pcie_pif/pcie_pif.c
index 63f33f2..4c56565 100644
--- a/app/pcie_pif/pcie_pif.c
+++ b/app/pcie_pif/pcie_pif.c
@@ -10,6 +10,8 @@
 #include <device.h>
 #include "cmd_module.h"
 
+struct device_pif_arc pif_arc;
+
 static struct cmd_module_ops cmd_ops[] = {
 	{	/* PCI module */
 		.setup = module_pci_setup,
@@ -29,6 +31,7 @@
 {
 	device_init_global_conf();
 	device_init_pci();
+	device_init_pif_arc();
 }
 
 void module_run_uncond_each_loop(void) {}
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 04437c6..f146b7b 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -19,11 +19,11 @@
 #include <d0_psoc_i2c_s_regs.h>
 #include <d0_parc_int_aggr_i2c_slv_comb_regs.h>
 #include <d0_parc_local_cfg_regs.h>
-
+#include <d0_pif_arc_regs.h>
 #include <d1_pcie_aux_regs.h>
 #include <d1_pcie_core_regs.h>
 #include <d1_pcie_dbi_sig_regs.h>
-
+#include <d1_pif_arc_regs.h>
 
 #include <gaudi3/gaudi3_boot_info.h>
 
@@ -330,6 +330,37 @@
 	ca53_cfg.arm_rst_control_reg = mmD0_CPU_CA53_CFG_ARM_RST_CONTROL;
 }
 
+void device_init_pif_arc(void)
+{
+	if (!hl_get_die_id()) {	/* DIE0 */
+		pif_arc.pf_flr_active = mmD0_PIF_ARC_PF_FLR_ACTIVE;
+		pif_arc.pf_flr_done = mmD0_PIF_ARC_PF_FLR_DONE;
+		pif_arc.inbound_term = mmD0_PIF_ARC_INBOUND_TERM;
+		pif_arc.crs = mmD0_PIF_ARC_CRS;
+		pif_arc.outbound_iso = mmD0_PIF_ARC_OUTBOUND_ISO;
+		pif_arc.outstand_trans = mmD0_PIF_ARC_OUTSTAND_TRANS;
+		pif_arc.pif2psoc_flr_active = mmD0_PIF_ARC_PIF2PSOC_FLR_ACTIVE;
+		pif_arc.psoc2pif_flr_req = mmD0_PIF_ARC_PSOC2PIF_FLR_REQ;
+		pif_arc.psoc2pif_flr_done = mmD0_PIF_ARC_PSOC2PIF_FLR_DONE;
+		pif_arc.psoc2pif_die_ready = mmD0_PIF_ARC_PSOC2PIF_DIE_READY;
+		pif_arc.timestamp_ctrl = mmD0_PIF_ARC_FLR_TIMESTAMP_CTRL;
+		pif_arc.timestamp_cnt = mmD0_PIF_ARC_FLR_TIMESTAMP_CNT;
+	} else {	/* DIE1 */
+		pif_arc.pf_flr_active = mmD1_PIF_ARC_PF_FLR_ACTIVE;
+		pif_arc.pf_flr_done = mmD1_PIF_ARC_PF_FLR_DONE;
+		pif_arc.inbound_term = mmD1_PIF_ARC_INBOUND_TERM;
+		pif_arc.crs = mmD1_PIF_ARC_CRS;
+		pif_arc.outbound_iso = mmD1_PIF_ARC_OUTBOUND_ISO;
+		pif_arc.outstand_trans = mmD1_PIF_ARC_OUTSTAND_TRANS;
+		pif_arc.pif2psoc_flr_active = mmD1_PIF_ARC_PIF2PSOC_FLR_ACTIVE;
+		pif_arc.psoc2pif_flr_req = mmD1_PIF_ARC_PSOC2PIF_FLR_REQ;
+		pif_arc.psoc2pif_flr_done = mmD1_PIF_ARC_PSOC2PIF_FLR_DONE;
+		pif_arc.psoc2pif_die_ready = mmD1_PIF_ARC_PSOC2PIF_DIE_READY;
+		pif_arc.timestamp_ctrl = mmD1_PIF_ARC_FLR_TIMESTAMP_CTRL;
+		pif_arc.timestamp_cnt = mmD1_PIF_ARC_FLR_TIMESTAMP_CNT;
+	}
+}
+
 void lkd_dev_specific_setup(void)
 {
 	cpu_boot_dev_sts_set(CPU_BOOT_DEV_STS_FW_HARD_RST_EN);
diff --git a/include/device.h b/include/device.h
index e100e12..cabd971 100644
--- a/include/device.h
+++ b/include/device.h
@@ -83,12 +83,28 @@
 	u32 arm_rst_control_reg;
 };
 
+struct device_pif_arc {
+	u32 pf_flr_active;
+	u32 pf_flr_done;
+	u32 inbound_term;
+	u32 crs;
+	u32 outbound_iso;
+	u32 outstand_trans;
+	u32 pif2psoc_flr_active;
+	u32 psoc2pif_flr_req;
+	u32 psoc2pif_flr_done;
+	u32 psoc2pif_die_ready;
+	u32 timestamp_ctrl;
+	u32 timestamp_cnt;
+};
+
 extern struct device_efuse efuse;
 extern struct device_uart uart;
 extern struct device_pci pci;
 extern struct device_i2c_s i2c_s;
 extern struct device_global_conf global_conf;
 extern struct device_ca53_cfg ca53_cfg;
+extern struct device_pif_arc pif_arc;
 
 void device_init_efuse(void);
 void device_init_uart(void);
@@ -96,3 +112,4 @@
 void device_init_i2c_s(void);
 void device_init_global_conf(void);
 void device_init_ca53_cfg(void);
+void device_init_pif_arc(void);
